libraries/core_libs/consensus/src/final_chain/rewards_stats.cpp

```diff
@@ -42,35 +44,46 @@ std::optional<addr_t> RewardsStats::getTransactionValidator(const trx_hash_t& tx
   return {found_tx->second};
 }
 
-bool RewardsStats::addVote(const Vote& vote) {
+bool RewardsStats::addVote(const std::shared_ptr<Vote>& vote, uint64_t min_weight) {
   // Set valid cert vote to validator
-  auto& validator_stats = validators_stats_[vote.getVoterAddr()];
-  assert(validator_stats.valid_cert_vote_ == false);
+  auto& validator_stats = validators_stats_[vote->getVoterAddr()];
+  assert(validator_stats.vote_weight_ == 0);
+  assert(vote->getWeight());
+
+  if (validator_stats.vote_weight_) {
+    return false;
+  }
 
-  validator_stats.valid_cert_vote_ = true;
-  total_unique_votes_count_++;
+  const auto weight = *vote->getWeight();
+  if (total_votes_weight_ > min_weight) {
+    bonus_votes_weight_ += weight;
+  }
 
+  total_votes_weight_ += weight;
+  validator_stats.vote_weight_ = weight;
   return true;
 }
 
-void RewardsStats::initStats(const PeriodData& sync_blk) {
+void RewardsStats::initStats(const PeriodData& sync_blk, uint64_t dpos_vote_count, uint32_t committee_size) {
   txs_validators_.reserve(sync_blk.transactions.size());
-
-  // TODO: use std::max(sync_blk.dag_blocks.size(), sync_blk.rewards_votes.size())
-  validators_stats_.reserve(sync_blk.dag_blocks.size());
+  validators_stats_.reserve(std::max(sync_blk.dag_blocks.size(), sync_blk.previous_block_cert_votes.size()));
 
   for (const auto& dag_block : sync_blk.dag_blocks) {
     const addr_t& dag_block_author = dag_block.getSender();
     for (const auto& tx_hash : dag_block.getTrxs()) {
       addTransaction(tx_hash, dag_block_author);
     }
   }
-
-  // TODO: add votes to rewards_stats
+  max_votes_weight_ = std::min<uint64_t>(committee_size, dpos_vote_count);
+  const uint64_t min_weight = max_votes_weight_ * 2 / 3 + 1;
+  for (const auto& vote : sync_blk.previous_block_cert_votes) {
+    addVote(vote, min_weight);
```

Comment:
why don't you create class member min_weight_ and set it few lines above instead of sending the same value hundreds of times into the addVote function ?

Reply:
because we are passing whole class to go and I did not want to add additional values that is not required by go